zombie-monitor/src/test/scala/com/broadinstitute/dsp/zombieMonitor/DeletedDiskCheckerSpec.scala

```diff
@@ -0,0 +1,261 @@
+package com.broadinstitute.dsp
+package zombieMonitor
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import cats.mtl.Ask
+import com.broadinstitute.dsp.Generators._
+import com.google.api.gax.grpc.GrpcStatusCode
+import com.google.api.gax.rpc.{InternalException, PermissionDeniedException}
+import com.google.cloud.compute.v1
+import io.grpc.Status
+import org.broadinstitute.dsde.workbench.google2.{DiskName, GoogleDiskService, ZoneName}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.mockito.ArgumentMatchers.{any, anyLong, anyString}
+import org.mockito.Mockito.{mock, never, verify, when}
+import org.scalatest.flatspec.AnyFlatSpec
+
+class DeletedDiskCheckerSpec extends AnyFlatSpec with CronJobsTestSuite {
+
+  var mockDbReader: DbReader[IO] = _
+  var mockCheckRunnerDeps: CheckRunnerDeps[IO] = _
+  var mockGoogleDiskService: GoogleDiskService[IO] = _
+  var mockDiskCheckerDeps: DiskCheckerDeps[IO] = _
+  var checker: CheckRunner[IO, Disk] = _
+
+  var billingDisabledException = new PermissionDeniedException(
+    new Exception("This API method requires billing to be enabled"),
+    GrpcStatusCode.of(Status.Code.PERMISSION_DENIED),
+    false
+  )
+  var computeEngineNotSetupException = new PermissionDeniedException(
+    new Exception("Compute Engine API has not been used"),
+    GrpcStatusCode.of(Status.Code.PERMISSION_DENIED),
+    false
+  )
+
+  val googleDisk: v1.Disk = com.google.cloud.compute.v1.Disk
+    .newBuilder()
+    .setName("disk-name")
+    .setSizeGb(100)
+    .setType("pd-standard")
+    .build()
+  def setupMocks(): Unit = {
+    mockDbReader = mock(classOf[DbReader[IO]])
+    mockCheckRunnerDeps = mock(classOf[CheckRunnerDeps[IO]])
+    mockGoogleDiskService = mock(classOf[GoogleDiskService[IO]])
+    mockDiskCheckerDeps = DiskCheckerDeps(mockCheckRunnerDeps, mockGoogleDiskService)
+    checker = DeletedDiskChecker.impl(mockDbReader, mockDiskCheckerDeps)
+  }
+
+  it should "call updateDiskStatus with error 'billing is disabled' (isDryRun = false)" in {
```

Comment:
the naming here was a little confusing to me at first, something like the following might be clearer?
`it should "updateDiskStatus when GCP returns a 'billing is disabled' error"`